internal/infrastructure/ceph/rbd.go

```diff
@@ -181,47 +222,130 @@ func applyDiffToBlockDevice(blockDevicePath string, diffFile io.Reader, fromSnap
 		return fmt.Errorf("block device size is smaller than diff size: %s", blockDevicePath)
 	}
 
-	if err = applyDiffDataRecords(blockDeviceFile, diffFileReader, 0, true); err != nil {
+	if err = applyDiffDataRecords(blockDeviceFile, diffFileReader, 0, true, false, 0, nil); err != nil {
 		return fmt.Errorf("failed to apply diff records to %s: %w", blockDevicePath, err)
 	}
 
 	return nil
 }
 
+//nolint:unparam // parameters are currently constant; remove once they become configurable.
 func applyDiffToRawImage(
 	rawImageFilePath string,
 	diffFile io.Reader,
 	fromSnapName,
 	toSnapName string,
-	expansionUnitSize uint64,
+	expansionUnitSize,
+	rawChecksumChunkSize uint64,
+	enableChecksumVerify bool,
 ) error {
 	diffFileReader, _, err := openDiffDataRecords(diffFile, fromSnapName, toSnapName)
 	if err != nil {
 		return fmt.Errorf("failed to open diff data records: %w", err)
 	}
 
-	// Open raw image file
 	var rawImgFile *os.File
-	if _, err := os.Stat(rawImageFilePath); os.IsNotExist(err) {
-		rawImgFile, err = os.OpenFile(rawImageFilePath, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0644)
+	var rawChecksumFile *os.File
+
+	rawStat, err := os.Stat(rawImageFilePath)
+	if err != nil && !os.IsNotExist(err) {
+		return fmt.Errorf("failed to stat raw image file: %w", err)
+	}
+
+	// When raw.img already exists check the chunk counts of raw.img and its
+	// checksum file. A smaller raw chunk count means the checksum was written
+	// and the process stopped while creating raw.img, so the next retry
+	// recreates both.
```

Comment:
Unfortunately, the following part of this comment is actually wrong:

> A smaller raw chunk count means the checksum was written
> and the process stopped while creating raw.img,

The smaller raw chunk _may_ mean the process stopped while writing the checksum.

Reply:
In addition, please add a comment to describe that this check may handle the case when the checksum file is extended (but the raw image file is not extended due to disruption) in L.443.